src/Compilers/CSharp/Portable/Parser/DocumentationCommentParser.cs

```diff
@@ -975,6 +977,39 @@ private IndexerMemberCrefSyntax ParseIndexerMemberCref()
             return SyntaxFactory.IndexerMemberCref(thisKeyword, parameters);
         }
 
+#nullable enable
+        /// <summary>
+        /// If we have `extension` (with optional type arguments) and parameter list and a dot, then we have an extension member cref.
+        /// Otherwise, we fall back to producing the same result as <see cref="ParseNameMemberCref"/>
+        /// </summary>
+        private MemberCrefSyntax ParsePossibleExtensionMemberCref()
+        {
+            Debug.Assert(CurrentToken.ContextualKind == SyntaxKind.ExtensionKeyword);
+
+            SyntaxToken identifierToken = EatToken();
+            TypeArgumentListSyntax? typeArguments = (CurrentToken.Kind == SyntaxKind.LessThanToken) ? ParseTypeArguments(typeArgumentsMustBeIdentifiers: true) : null;
+            CrefParameterListSyntax? parameters = (CurrentToken.Kind == SyntaxKind.OpenParenToken) ? ParseCrefParameterList() : null;
+
+            if (parameters is null || CurrentToken.Kind != SyntaxKind.DotToken)
+            {
+                SimpleNameSyntax name = typeArguments is not null
+                    ? SyntaxFactory.GenericName(identifierToken, typeArguments)
+                    : SyntaxFactory.IdentifierName(identifierToken);
+
+                return SyntaxFactory.NameMemberCref(name, parameters);
+            }
+
+            SyntaxToken dotToken = EatToken(SyntaxKind.DotToken);
+            MemberCrefSyntax member = ParseMemberCref();
+            if (member is ExtensionMemberCrefSyntax)
```

Comment:
>if (member is ExtensionMemberCrefSyntax) [](http://example.com/codeflow?start=12&length=40)

My preference would be to pass a flag to ```ParseMemberCref``` instructing to ignore the extension case, but I can live with this approach as well.

Reply:
I'm leaning to keep current approach (parse and report an error)